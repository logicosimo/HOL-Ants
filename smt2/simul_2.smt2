(declare-datatype Position ((P0) (P1) (P2) (P3) (P4)))
(declare-const a Int)
(declare-const a_1 Int)
(declare-const a_2 Int)
(declare-const a_3 Int)
(declare-const a_4 Int)
(declare-const a_5 Int)
(declare-const a_6 Int)
(declare-const a_7 Int)
(declare-const a_8 Int)
(declare-const a_9 Int)
(declare-const a_10 Int)
(declare-const a_11 Int)
(declare-const a_12 Int)
(declare-const a_13 Int)
(declare-const a_14 Int)
(declare-const a_15 Int)
(declare-const a_16 Int)
(declare-const a_17 Int)
(declare-const a1 Position)
(declare-const a2 Position)
(declare-const c Int)
(declare-const c_1 Int)
(declare-const c_2 Int)
(declare-const c_3 Int)
(declare-const c_4 Int)
(declare-const c_5 Int)
(declare-const c_6 Int)
(declare-const c_7 Int)
(declare-const c_8 Int)
(declare-const d1 Bool)
(declare-const d2 Bool)
(declare-const dir Bool)
(declare-const dir_1 Bool)
(declare-const dir_2 Bool)
(declare-const dir_3 Bool)
(declare-const dir_4 Bool)
(declare-const dir_5 Bool)
(declare-const dir_6 Bool)
(declare-const dir_7 Bool)
(declare-const dir_8 Bool)
(declare-const dir_9 Bool)
(declare-const dir_10 Bool)
(declare-const dir_11 Bool)
(declare-const dir_12 Bool)
(declare-const dir_13 Bool)
(declare-const dir_14 Bool)
(declare-const dir_15 Bool)
(declare-const dir_16 Bool)
(declare-const dir_17 Bool)
(declare-const pos Position)
(declare-const pos_1 Position)
(declare-const pos_2 Position)
(declare-const pos_3 Position)
(declare-const pos_4 Position)
(declare-const pos_5 Position)
(declare-const pos_6 Position)
(declare-const pos_7 Position)
(declare-const pos_8 Position)
(declare-const pos_9 Position)
(declare-const pos_10 Position)
(declare-const pos_11 Position)
(declare-const pos_12 Position)
(declare-const pos_13 Position)
(declare-const pos_14 Position)
(declare-const pos_15 Position)
(declare-const pos_16 Position)
(declare-const pos_17 Position)
(declare-const s1 Int)
(declare-const s2 Int)
(declare-const s3 Int)
(assert (>= a 0))
(assert (>= a_1 0))
(assert (>= a_2 0))
(assert (>= a_3 0))
(assert (>= a_4 0))
(assert (>= a_5 0))
(assert (>= a_6 0))
(assert (>= a_7 0))
(assert (>= a_8 0))
(assert (>= a_9 0))
(assert (>= a_10 0))
(assert (>= a_11 0))
(assert (>= a_12 0))
(assert (>= a_13 0))
(assert (>= a_14 0))
(assert (>= a_15 0))
(assert (>= a_16 0))
(assert (>= a_17 0))
(assert (>= c 0))
(assert (>= c_1 0))
(assert (>= c_2 0))
(assert (>= c_3 0))
(assert (>= c_4 0))
(assert (>= c_5 0))
(assert (>= c_6 0))
(assert (>= c_7 0))
(assert (>= c_8 0))
(assert (>= s1 0))
(assert (>= s2 0))
(assert (>= s3 0))
(assert
 (and (= a_16 (+ 0 (ite (= P0 P1) 1 0) 1))
  (= c_8 (+ 0 (ite (= P0 P3) 1 0) (ite (= P1 P3) 1 0)))
  (= a_17 (+ 0 (ite (= P0 P2) 1 0) (ite (= P1 P2) 1 0)))
  (=> (= P0 P1) (and (= pos_16 P4) dir_16))
  (=> (= P0 P2) (and (= pos_16 P3) dir_16))
  (=> (= P0 P3) (and (= pos_16 P4) dir_16)) (or (= pos_16 P1) (= pos_16 P2))
  dir_16 (=> (= pos_16 P1) (<= 0 0)) (=> (= pos_16 P2) (<= 0 0))
  (=> (= P0 P4) (and (or (= pos_16 P1) (= pos_16 P3)) (not dir_16)))
  (=> (and (= P0 P4) (= pos_16 P1)) (<= 0 0))
  (=> (and (= P0 P4) (= pos_16 P3)) (<= 0 0)) (= pos_17 P0) (not dir_17)
  (=> (= P1 P2) (and (= pos_17 P0) (not dir_17)))
  (=> (= P1 P3) (and (= pos_17 P2) (not dir_17)))
  (=> (= P1 P0) (and (or (= pos_17 P1) (= pos_17 P2)) dir_17))
  (=> (and (= P1 P0) (= pos_17 P1)) (<= 0 0))
  (=> (and (= P1 P0) (= pos_17 P2)) (<= 0 0))
  (=> (= P1 P4) (and (or (= pos_17 P1) (= pos_17 P3)) (not dir_17)))
  (=> (and (= P1 P4) (= pos_17 P1)) (<= 0 0))
  (=> (and (= P1 P4) (= pos_17 P3)) (<= 0 0))
  (= a_14 (+ a_16 (ite (= pos_16 P1) 1 0) (ite (= pos_17 P1) 1 0)))
  (= c_7 (+ c_8 (ite (= pos_16 P3) 1 0) (ite (= pos_17 P3) 1 0)))
  (= a_15 (+ a_17 (ite (= pos_16 P2) 1 0) (ite (= pos_17 P2) 1 0)))
  (=> (= pos_16 P1) (and (= pos_14 (ite dir_16 P4 P0)) (iff dir_14 dir_16)))
  (=> (= pos_16 P2) (and (= pos_14 (ite dir_16 P3 P0)) (iff dir_14 dir_16)))
  (=> (= pos_16 P3) (and (= pos_14 (ite dir_16 P4 P2)) (iff dir_14 dir_16)))
  (=> (= pos_16 P0) (and (or (= pos_14 P1) (= pos_14 P2)) dir_14))
  (=> (and (= pos_16 P0) (= pos_14 P1)) (<= a_17 a_16))
  (=> (and (= pos_16 P0) (= pos_14 P2)) (<= a_16 a_17))
  (=> (= pos_16 P4) (and (or (= pos_14 P1) (= pos_14 P3)) (not dir_14)))
  (=> (and (= pos_16 P4) (= pos_14 P1)) (<= c_8 a_16))
  (=> (and (= pos_16 P4) (= pos_14 P3)) (<= a_16 c_8))
  (=> (= pos_17 P1) (and (= pos_15 (ite dir_17 P4 P0)) (iff dir_15 dir_17)))
  (=> (= pos_17 P2) (and (= pos_15 (ite dir_17 P3 P0)) (iff dir_15 dir_17)))
  (=> (= pos_17 P3) (and (= pos_15 (ite dir_17 P4 P2)) (iff dir_15 dir_17)))
  (=> (= pos_17 P0) (and (or (= pos_15 P1) (= pos_15 P2)) dir_15))
  (=> (and (= pos_17 P0) (= pos_15 P1)) (<= a_17 a_16))
  (=> (and (= pos_17 P0) (= pos_15 P2)) (<= a_16 a_17))
  (=> (= pos_17 P4) (and (or (= pos_15 P1) (= pos_15 P3)) (not dir_15)))
  (=> (and (= pos_17 P4) (= pos_15 P1)) (<= c_8 a_16))
  (=> (and (= pos_17 P4) (= pos_15 P3)) (<= a_16 c_8))
  (= a_12 (+ a_14 (ite (= pos_14 P1) 1 0) (ite (= pos_15 P1) 1 0)))
  (= c_6 (+ c_7 (ite (= pos_14 P3) 1 0) (ite (= pos_15 P3) 1 0)))
  (= a_13 (+ a_15 (ite (= pos_14 P2) 1 0) (ite (= pos_15 P2) 1 0)))
  (=> (= pos_14 P1) (and (= pos_12 (ite dir_14 P4 P0)) (iff dir_12 dir_14)))
  (=> (= pos_14 P2) (and (= pos_12 (ite dir_14 P3 P0)) (iff dir_12 dir_14)))
  (=> (= pos_14 P3) (and (= pos_12 (ite dir_14 P4 P2)) (iff dir_12 dir_14)))
  (=> (= pos_14 P0) (and (or (= pos_12 P1) (= pos_12 P2)) dir_12))
  (=> (and (= pos_14 P0) (= pos_12 P1)) (<= a_15 a_14))
  (=> (and (= pos_14 P0) (= pos_12 P2)) (<= a_14 a_15))
  (=> (= pos_14 P4) (and (or (= pos_12 P1) (= pos_12 P3)) (not dir_12)))
  (=> (and (= pos_14 P4) (= pos_12 P1)) (<= c_7 a_14))
  (=> (and (= pos_14 P4) (= pos_12 P3)) (<= a_14 c_7))
  (=> (= pos_15 P1) (and (= pos_13 (ite dir_15 P4 P0)) (iff dir_13 dir_15)))
  (=> (= pos_15 P2) (and (= pos_13 (ite dir_15 P3 P0)) (iff dir_13 dir_15)))
  (=> (= pos_15 P3) (and (= pos_13 (ite dir_15 P4 P2)) (iff dir_13 dir_15)))
  (=> (= pos_15 P0) (and (or (= pos_13 P1) (= pos_13 P2)) dir_13))
  (=> (and (= pos_15 P0) (= pos_13 P1)) (<= a_15 a_14))
  (=> (and (= pos_15 P0) (= pos_13 P2)) (<= a_14 a_15))
  (=> (= pos_15 P4) (and (or (= pos_13 P1) (= pos_13 P3)) (not dir_13)))
  (=> (and (= pos_15 P4) (= pos_13 P1)) (<= c_7 a_14))
  (=> (and (= pos_15 P4) (= pos_13 P3)) (<= a_14 c_7))
  (= a_10 (+ a_12 (ite (= pos_12 P1) 1 0) (ite (= pos_13 P1) 1 0)))
  (= c_5 (+ c_6 (ite (= pos_12 P3) 1 0) (ite (= pos_13 P3) 1 0)))
  (= a_11 (+ a_13 (ite (= pos_12 P2) 1 0) (ite (= pos_13 P2) 1 0)))
  (=> (= pos_12 P1) (and (= pos_10 (ite dir_12 P4 P0)) (iff dir_10 dir_12)))
  (=> (= pos_12 P2) (and (= pos_10 (ite dir_12 P3 P0)) (iff dir_10 dir_12)))
  (=> (= pos_12 P3) (and (= pos_10 (ite dir_12 P4 P2)) (iff dir_10 dir_12)))
  (=> (= pos_12 P0) (and (or (= pos_10 P1) (= pos_10 P2)) dir_10))
  (=> (and (= pos_12 P0) (= pos_10 P1)) (<= a_13 a_12))
  (=> (and (= pos_12 P0) (= pos_10 P2)) (<= a_12 a_13))
  (=> (= pos_12 P4) (and (or (= pos_10 P1) (= pos_10 P3)) (not dir_10)))
  (=> (and (= pos_12 P4) (= pos_10 P1)) (<= c_6 a_12))
  (=> (and (= pos_12 P4) (= pos_10 P3)) (<= a_12 c_6))
  (=> (= pos_13 P1) (and (= pos_11 (ite dir_13 P4 P0)) (iff dir_11 dir_13)))
  (=> (= pos_13 P2) (and (= pos_11 (ite dir_13 P3 P0)) (iff dir_11 dir_13)))
  (=> (= pos_13 P3) (and (= pos_11 (ite dir_13 P4 P2)) (iff dir_11 dir_13)))
  (=> (= pos_13 P0) (and (or (= pos_11 P1) (= pos_11 P2)) dir_11))
  (=> (and (= pos_13 P0) (= pos_11 P1)) (<= a_13 a_12))
  (=> (and (= pos_13 P0) (= pos_11 P2)) (<= a_12 a_13))
  (=> (= pos_13 P4) (and (or (= pos_11 P1) (= pos_11 P3)) (not dir_11)))
  (=> (and (= pos_13 P4) (= pos_11 P1)) (<= c_6 a_12))
  (=> (and (= pos_13 P4) (= pos_11 P3)) (<= a_12 c_6))
  (= a_8 (+ a_10 (ite (= pos_10 P1) 1 0) (ite (= pos_11 P1) 1 0)))
  (= c_4 (+ c_5 (ite (= pos_10 P3) 1 0) (ite (= pos_11 P3) 1 0)))
  (= a_9 (+ a_11 (ite (= pos_10 P2) 1 0) (ite (= pos_11 P2) 1 0)))
  (=> (= pos_10 P1) (and (= pos_8 (ite dir_10 P4 P0)) (iff dir_8 dir_10)))
  (=> (= pos_10 P2) (and (= pos_8 (ite dir_10 P3 P0)) (iff dir_8 dir_10)))
  (=> (= pos_10 P3) (and (= pos_8 (ite dir_10 P4 P2)) (iff dir_8 dir_10)))
  (=> (= pos_10 P0) (and (or (= pos_8 P1) (= pos_8 P2)) dir_8))
  (=> (and (= pos_10 P0) (= pos_8 P1)) (<= a_11 a_10))
  (=> (and (= pos_10 P0) (= pos_8 P2)) (<= a_10 a_11))
  (=> (= pos_10 P4) (and (or (= pos_8 P1) (= pos_8 P3)) (not dir_8)))
  (=> (and (= pos_10 P4) (= pos_8 P1)) (<= c_5 a_10))
  (=> (and (= pos_10 P4) (= pos_8 P3)) (<= a_10 c_5))
  (=> (= pos_11 P1) (and (= pos_9 (ite dir_11 P4 P0)) (iff dir_9 dir_11)))
  (=> (= pos_11 P2) (and (= pos_9 (ite dir_11 P3 P0)) (iff dir_9 dir_11)))
  (=> (= pos_11 P3) (and (= pos_9 (ite dir_11 P4 P2)) (iff dir_9 dir_11)))
  (=> (= pos_11 P0) (and (or (= pos_9 P1) (= pos_9 P2)) dir_9))
  (=> (and (= pos_11 P0) (= pos_9 P1)) (<= a_11 a_10))
  (=> (and (= pos_11 P0) (= pos_9 P2)) (<= a_10 a_11))
  (=> (= pos_11 P4) (and (or (= pos_9 P1) (= pos_9 P3)) (not dir_9)))
  (=> (and (= pos_11 P4) (= pos_9 P1)) (<= c_5 a_10))
  (=> (and (= pos_11 P4) (= pos_9 P3)) (<= a_10 c_5))
  (= a_6 (+ a_8 (ite (= pos_8 P1) 1 0) (ite (= pos_9 P1) 1 0)))
  (= c_3 (+ c_4 (ite (= pos_8 P3) 1 0) (ite (= pos_9 P3) 1 0)))
  (= a_7 (+ a_9 (ite (= pos_8 P2) 1 0) (ite (= pos_9 P2) 1 0)))
  (=> (= pos_8 P1) (and (= pos_6 (ite dir_8 P4 P0)) (iff dir_6 dir_8)))
  (=> (= pos_8 P2) (and (= pos_6 (ite dir_8 P3 P0)) (iff dir_6 dir_8)))
  (=> (= pos_8 P3) (and (= pos_6 (ite dir_8 P4 P2)) (iff dir_6 dir_8)))
  (=> (= pos_8 P0) (and (or (= pos_6 P1) (= pos_6 P2)) dir_6))
  (=> (and (= pos_8 P0) (= pos_6 P1)) (<= a_9 a_8))
  (=> (and (= pos_8 P0) (= pos_6 P2)) (<= a_8 a_9))
  (=> (= pos_8 P4) (and (or (= pos_6 P1) (= pos_6 P3)) (not dir_6)))
  (=> (and (= pos_8 P4) (= pos_6 P1)) (<= c_4 a_8))
  (=> (and (= pos_8 P4) (= pos_6 P3)) (<= a_8 c_4))
  (=> (= pos_9 P1) (and (= pos_7 (ite dir_9 P4 P0)) (iff dir_7 dir_9)))
  (=> (= pos_9 P2) (and (= pos_7 (ite dir_9 P3 P0)) (iff dir_7 dir_9)))
  (=> (= pos_9 P3) (and (= pos_7 (ite dir_9 P4 P2)) (iff dir_7 dir_9)))
  (=> (= pos_9 P0) (and (or (= pos_7 P1) (= pos_7 P2)) dir_7))
  (=> (and (= pos_9 P0) (= pos_7 P1)) (<= a_9 a_8))
  (=> (and (= pos_9 P0) (= pos_7 P2)) (<= a_8 a_9))
  (=> (= pos_9 P4) (and (or (= pos_7 P1) (= pos_7 P3)) (not dir_7)))
  (=> (and (= pos_9 P4) (= pos_7 P1)) (<= c_4 a_8))
  (=> (and (= pos_9 P4) (= pos_7 P3)) (<= a_8 c_4))
  (= a_4 (+ a_6 (ite (= pos_6 P1) 1 0) (ite (= pos_7 P1) 1 0)))
  (= c_2 (+ c_3 (ite (= pos_6 P3) 1 0) (ite (= pos_7 P3) 1 0)))
  (= a_5 (+ a_7 (ite (= pos_6 P2) 1 0) (ite (= pos_7 P2) 1 0)))
  (=> (= pos_6 P1) (and (= pos_4 (ite dir_6 P4 P0)) (iff dir_4 dir_6)))
  (=> (= pos_6 P2) (and (= pos_4 (ite dir_6 P3 P0)) (iff dir_4 dir_6)))
  (=> (= pos_6 P3) (and (= pos_4 (ite dir_6 P4 P2)) (iff dir_4 dir_6)))
  (=> (= pos_6 P0) (and (or (= pos_4 P1) (= pos_4 P2)) dir_4))
  (=> (and (= pos_6 P0) (= pos_4 P1)) (<= a_7 a_6))
  (=> (and (= pos_6 P0) (= pos_4 P2)) (<= a_6 a_7))
  (=> (= pos_6 P4) (and (or (= pos_4 P1) (= pos_4 P3)) (not dir_4)))
  (=> (and (= pos_6 P4) (= pos_4 P1)) (<= c_3 a_6))
  (=> (and (= pos_6 P4) (= pos_4 P3)) (<= a_6 c_3))
  (=> (= pos_7 P1) (and (= pos_5 (ite dir_7 P4 P0)) (iff dir_5 dir_7)))
  (=> (= pos_7 P2) (and (= pos_5 (ite dir_7 P3 P0)) (iff dir_5 dir_7)))
  (=> (= pos_7 P3) (and (= pos_5 (ite dir_7 P4 P2)) (iff dir_5 dir_7)))
  (=> (= pos_7 P0) (and (or (= pos_5 P1) (= pos_5 P2)) dir_5))
  (=> (and (= pos_7 P0) (= pos_5 P1)) (<= a_7 a_6))
  (=> (and (= pos_7 P0) (= pos_5 P2)) (<= a_6 a_7))
  (=> (= pos_7 P4) (and (or (= pos_5 P1) (= pos_5 P3)) (not dir_5)))
  (=> (and (= pos_7 P4) (= pos_5 P1)) (<= c_3 a_6))
  (=> (and (= pos_7 P4) (= pos_5 P3)) (<= a_6 c_3))
  (= a_2 (+ a_4 (ite (= pos_4 P1) 1 0) (ite (= pos_5 P1) 1 0)))
  (= c_1 (+ c_2 (ite (= pos_4 P3) 1 0) (ite (= pos_5 P3) 1 0)))
  (= a_3 (+ a_5 (ite (= pos_4 P2) 1 0) (ite (= pos_5 P2) 1 0)))
  (=> (= pos_4 P1) (and (= pos_2 (ite dir_4 P4 P0)) (iff dir_2 dir_4)))
  (=> (= pos_4 P2) (and (= pos_2 (ite dir_4 P3 P0)) (iff dir_2 dir_4)))
  (=> (= pos_4 P3) (and (= pos_2 (ite dir_4 P4 P2)) (iff dir_2 dir_4)))
  (=> (= pos_4 P0) (and (or (= pos_2 P1) (= pos_2 P2)) dir_2))
  (=> (and (= pos_4 P0) (= pos_2 P1)) (<= a_5 a_4))
  (=> (and (= pos_4 P0) (= pos_2 P2)) (<= a_4 a_5))
  (=> (= pos_4 P4) (and (or (= pos_2 P1) (= pos_2 P3)) (not dir_2)))
  (=> (and (= pos_4 P4) (= pos_2 P1)) (<= c_2 a_4))
  (=> (and (= pos_4 P4) (= pos_2 P3)) (<= a_4 c_2))
  (=> (= pos_5 P1) (and (= pos_3 (ite dir_5 P4 P0)) (iff dir_3 dir_5)))
  (=> (= pos_5 P2) (and (= pos_3 (ite dir_5 P3 P0)) (iff dir_3 dir_5)))
  (=> (= pos_5 P3) (and (= pos_3 (ite dir_5 P4 P2)) (iff dir_3 dir_5)))
  (=> (= pos_5 P0) (and (or (= pos_3 P1) (= pos_3 P2)) dir_3))
  (=> (and (= pos_5 P0) (= pos_3 P1)) (<= a_5 a_4))
  (=> (and (= pos_5 P0) (= pos_3 P2)) (<= a_4 a_5))
  (=> (= pos_5 P4) (and (or (= pos_3 P1) (= pos_3 P3)) (not dir_3)))
  (=> (and (= pos_5 P4) (= pos_3 P1)) (<= c_2 a_4))
  (=> (and (= pos_5 P4) (= pos_3 P3)) (<= a_4 c_2))
  (= a (+ a_2 (ite (= pos_2 P1) 1 0) (ite (= pos_3 P1) 1 0)))
  (= c (+ c_1 (ite (= pos_2 P3) 1 0) (ite (= pos_3 P3) 1 0)))
  (= a_1 (+ a_3 (ite (= pos_2 P2) 1 0) (ite (= pos_3 P2) 1 0)))
  (=> (= pos_2 P1) (and (= pos (ite dir_2 P4 P0)) (iff dir dir_2)))
  (=> (= pos_2 P2) (and (= pos (ite dir_2 P3 P0)) (iff dir dir_2)))
  (=> (= pos_2 P3) (and (= pos (ite dir_2 P4 P2)) (iff dir dir_2)))
  (=> (= pos_2 P0) (and (or (= pos P1) (= pos P2)) dir))
  (=> (and (= pos_2 P0) (= pos P1)) (<= a_3 a_2))
  (=> (and (= pos_2 P0) (= pos P2)) (<= a_2 a_3))
  (=> (= pos_2 P4) (and (or (= pos P1) (= pos P3)) (not dir)))
  (=> (and (= pos_2 P4) (= pos P1)) (<= c_1 a_2))
  (=> (and (= pos_2 P4) (= pos P3)) (<= a_2 c_1))
  (=> (= pos_3 P1) (and (= pos_1 (ite dir_3 P4 P0)) (iff dir_1 dir_3)))
  (=> (= pos_3 P2) (and (= pos_1 (ite dir_3 P3 P0)) (iff dir_1 dir_3)))
  (=> (= pos_3 P3) (and (= pos_1 (ite dir_3 P4 P2)) (iff dir_1 dir_3)))
  (=> (= pos_3 P0) (and (or (= pos_1 P1) (= pos_1 P2)) dir_1))
  (=> (and (= pos_3 P0) (= pos_1 P1)) (<= a_3 a_2))
  (=> (and (= pos_3 P0) (= pos_1 P2)) (<= a_2 a_3))
  (=> (= pos_3 P4) (and (or (= pos_1 P1) (= pos_1 P3)) (not dir_1)))
  (=> (and (= pos_3 P4) (= pos_1 P1)) (<= c_1 a_2))
  (=> (and (= pos_3 P4) (= pos_1 P3)) (<= a_2 c_1))
  (= s1 (+ a (ite (= pos P1) 1 0) (ite (= pos_1 P1) 1 0)))
  (= s3 (+ c (ite (= pos P3) 1 0) (ite (= pos_1 P3) 1 0)))
  (= s2 (+ a_1 (ite (= pos P2) 1 0) (ite (= pos_1 P2) 1 0)))
  (=> (= pos P1) (and (= a1 (ite dir P4 P0)) (iff d1 dir)))
  (=> (= pos P2) (and (= a1 (ite dir P3 P0)) (iff d1 dir)))
  (=> (= pos P3) (and (= a1 (ite dir P4 P2)) (iff d1 dir)))
  (=> (= pos P0) (and (or (= a1 P1) (= a1 P2)) d1))
  (=> (and (= pos P0) (= a1 P1)) (<= a_1 a))
  (=> (and (= pos P0) (= a1 P2)) (<= a a_1))
  (=> (= pos P4) (and (or (= a1 P1) (= a1 P3)) (not d1)))
  (=> (and (= pos P4) (= a1 P1)) (<= c a))
  (=> (and (= pos P4) (= a1 P3)) (<= a c))
  (=> (= pos_1 P1) (and (= a2 (ite dir_1 P4 P0)) (iff d2 dir_1)))
  (=> (= pos_1 P2) (and (= a2 (ite dir_1 P3 P0)) (iff d2 dir_1)))
  (=> (= pos_1 P3) (and (= a2 (ite dir_1 P4 P2)) (iff d2 dir_1)))
  (=> (= pos_1 P0) (and (or (= a2 P1) (= a2 P2)) d2))
  (=> (and (= pos_1 P0) (= a2 P1)) (<= a_1 a))
  (=> (and (= pos_1 P0) (= a2 P2)) (<= a a_1))
  (=> (= pos_1 P4) (and (or (= a2 P1) (= a2 P3)) (not d2)))
  (=> (and (= pos_1 P4) (= a2 P1)) (<= c a))
  (=> (and (= pos_1 P4) (= a2 P3)) (<= a c))))
(check-sat)
(get-value
 (a a_1 a_2 a_3 a_4 a_5 a_6 a_7 a_8 a_9 a_10 a_11 a_12 a_13 a_14 a_15 a_16
  a_17 a1 a2 c c_1 c_2 c_3 c_4 c_5 c_6 c_7 c_8 d1 d2 dir dir_1 dir_2 dir_3
  dir_4 dir_5 dir_6 dir_7 dir_8 dir_9 dir_10 dir_11 dir_12 dir_13 dir_14
  dir_15 dir_16 dir_17 pos pos_1 pos_2 pos_3 pos_4 pos_5 pos_6 pos_7 pos_8
  pos_9 pos_10 pos_11 pos_12 pos_13 pos_14 pos_15 pos_16 pos_17 s1 s2 s3))
