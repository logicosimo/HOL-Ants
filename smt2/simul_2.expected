sat
((a 8)
 (a_1 1)
 (a_2 6)
 (a_3 1)
 (a_4 6)
 (a_5 1)
 (a_6 4)
 (a_7 1)
 (a_8 4)
 (a_9 1)
 (a_10 2)
 (a_11 1)
 (a_12 2)
 (a_13 1)
 (a_14 1)
 (a_15 1)
 (a_16 1)
 (a_17 0)
 (a1 P1)
 (a2 P1)
 (c 1)
 (c_1 1)
 (c_2 1)
 (c_3 1)
 (c_4 1)
 (c_5 1)
 (c_6 1)
 (c_7 0)
 (c_8 0)
 (d1 true)
 (d2 true)
 (dir false)
 (dir_1 false)
 (dir_2 false)
 (dir_3 false)
 (dir_4 true)
 (dir_5 true)
 (dir_6 true)
 (dir_7 true)
 (dir_8 false)
 (dir_9 false)
 (dir_10 false)
 (dir_11 false)
 (dir_12 true)
 (dir_13 true)
 (dir_14 true)
 (dir_15 true)
 (dir_16 true)
 (dir_17 false)
 (pos P0)
 (pos_1 P0)
 (pos_2 P1)
 (pos_3 P1)
 (pos_4 P4)
 (pos_5 P4)
 (pos_6 P1)
 (pos_7 P1)
 (pos_8 P0)
 (pos_9 P0)
 (pos_10 P1)
 (pos_11 P1)
 (pos_12 P4)
 (pos_13 P4)
 (pos_14 P3)
 (pos_15 P1)
 (pos_16 P2)
 (pos_17 P0)
 (s1 8)
 (s2 1)
 (s3 1))
